decidim-debates/lib/decidim/debates/engine.rb

```diff
@@ -97,6 +97,37 @@ class Engine < ::Rails::Engine
         end
       end
 
+      initializer "decidim_debates.register_events" do
+        config.to_prepare do
+          ActiveSupport::Notifications.subscribe("decidim.debates.create_debate:after") do |_event_name, data|
+            Decidim::EventsManager.publish(
```

Comment:
Moving these outside of the command is still bothering me a bit because we should use the same convention everywhere. E.g. with meetings and comments, the events are still handled within the commands.

And I'm also not 100% sure if we should handle these outside of the commands since this would move a lot of complexity and logic to the events handling which is currently nicely wrapped inside the commands.

I understand the intention here but I'm just not sure if this is a good idea (+ it's not consistent over all commands).

Reply:
The main idea was to start moving decoupling the events from the commands ... IMO, a command should dispatch an event with some payload, then hook into from other place with the required events.